Improve Validations and Copy

vkoves opened the pull request:
This fixes validations not working on update-email and update-password. This also updates all copy to indicate that the password and email being entered are the new password and email, and makes update email consistent with update password by titling it change password.

Here are the validations if all form fields are blank:
![screenshot from 2017-07-28 12-08-17](https://user-images.githubusercontent.com/3187531/28728234-756a4cd0-738d-11e7-9615-83380c496a4e.png)
![screenshot from 2017-07-28 12-08-46](https://user-images.githubusercontent.com/3187531/28728249-84d657c2-738d-11e7-96f2-fcbbcb32995a.png)

Commit:
Fixed update-email validation and renamed for clarity
The update-email validation was called an email validation, but since it validates form fields specific to the update-email component, it has been renamed. A few last eslint errors on the email validator were also remedied, as well as errors being improved on the update-email component to make them properly reflect the form and to ensure validations are run before a submission is processed.

Commit:
Fixed validations for update-password and renamed them

Commit:
Updated copy on update email page to make it change email
This makes it more consistent with the change password page

conradlz commented:
I think the delete account should have a validation for cannot be blank so it would be consistent with the others.

Commit:
Added validation to delete-account
This tells the user to enter their password if they do not when submitting the form

vkoves commented:
@PrinceCornNM - I've added validation to `delete-account` like you suggested. Here's a screenshot:

![screenshot from 2017-07-28 12-23-39](https://user-images.githubusercontent.com/3187531/28728822-bb10f11a-738f-11e7-8236-f062c3927ceb.png)

conradlz reviewed (approved):
Looks Good
